### PR TITLE
Add an OEP-2 compliant openedx.yaml file

### PR DESCRIPTION
**Pull request by cpennington**
This adds an `openedx.yaml` file, as described by OEP-2:
http://open-edx-proposals.readthedocs.io/en/latest/oeps/oep-0002.html

The data in this file was transformed from the contents of
edx/repo-tools-data:repos.yaml

### CODE REVIEW

**Comment by cpennington** [comment]
@edx/devops: Labeling this repo as yours.

File: openedx.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# This file describes this Open edX repo, as described in OEP-2:
+# http://open-edx-proposals.readthedocs.io/en/latest/oeps/oep-0002.html#specification
+
+nick: conf
```
Comment (jibsheet) [inline openedx.yaml:4]: Can we have config, not conf?

File: openedx.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+oeps: {}
+openedx-release: {ref: master}
```
Comment (jibsheet) [inline openedx.yaml:6]: Having trouble finding the doc for this in the spec, but the openedx-release doesn't actually come from master unless you mean edx-platform's release branch.

File: openedx.yaml (hunk 1 above)
Comment (cpennington) [reply openedx.yaml:4]: This is for some open-source team tooling that uses very short textual labels.

File: openedx.yaml (hunk 2 above)
Comment (nedbat) [reply openedx.yaml:6]: This means, when it is time to tag an Open edX release, we are going to tag the master branch in edx/configuration.

File: openedx.yaml (hunk 1 above)
Comment (nedbat) [reply openedx.yaml:4]: @cpennington seems like we should add some mention of these keys to OEP-2.

Comment (jibsheet) [reply openedx.yaml:4]: That would be grand.

Comment (cpennington) [reply openedx.yaml:4]: Ok. I'd almost rather have them as part of an OEP describing how we do named releases, though.

Comment (cpennington) [reply openedx.yaml:4]: Ok, I now have a PR open to add docs of those keys to OEP-2.

**Comment by cpennington** [comment]
@jibsheet: ok to merge?

File: openedx.yaml (hunk 1 above)
Comment (feanil) [reply openedx.yaml:4]: Any reason why it can't just be 'configuration'?  Seems like a small enough word.

Comment (cpennington) [reply openedx.yaml:4]: See the definition here: https://github.com/edx/open-edx-proposals/pull/18/files#diff-13c9b7b8133d4bcf6ab907c7f0065187R53

**Comment by jibsheet** [comment]
I would love Open Source to fix their tools that requires conf as a nickname, is there a ticket for that?
But this is fine to merge 👍

**Comment by cpennington** [comment]
Not sure why everyone is getting so hung up on the short nickname...

**Pull request merged**